js/src/Ice/AsyncResult.d.ts

```diff
@@ -4,20 +4,56 @@
 
 declare module "ice" {
     namespace Ice {
-        class AsyncResult<T> extends AsyncResultBase<T> {
-            constructor(
-                communicator: Communicator,
-                operation: string,
-                connection: Connection,
-                proxy: ObjectPrx,
-                adapter: ObjectAdapter,
-                completed: (result: AsyncResult<T>) => void,
-            );
+        /**
```

Comment:
Not directly related to this PR - do we need this `AsyncResult<T>` class?

In particular, is deriving from Promise the standard way to implement cancellation in JS?

Reply:
There is no standard cancelation mechanism for async in JavaScript.

Reply:
There is a stage 1 proposal https://github.com/tc39/proposal-cancellation?tab=readme-ov-file#examples